src/main/java/pl/myCompany/transaction/TransactionController.java

```diff
@@ -0,0 +1,92 @@
+package pl.myCompany.transaction;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.ResponseEntity.BodyBuilder;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@Slf4j
+@RestController
+public class TransactionController {
+
+  private TransactionService transactionService;
+
+  public TransactionController(TransactionService transactionService) {
+    this.transactionService = transactionService;
+  }
+
+  @GetMapping("/transactions")
+  public ResponseEntity<List<Transaction>> getTransactions() {
+    Iterable<Transaction> transactionsIterable = transactionService.getTransactions();
+    Iterator<Transaction> transactionIterator = transactionsIterable.iterator();
+    List<Transaction> transactionList = convertIterableToCollectionList(transactionIterator);
+    return ResponseEntity.ok(transactionList);
+  }
+
+  private List<Transaction> convertIterableToCollectionList(Iterator<Transaction> transactions) {
+    List<Transaction> transactionList = new ArrayList<>();
+    transactions.forEachRemaining(transactionList::add);
+    return transactionList;
+  }
+
+  @GetMapping("/transaction/{id}")
+  public ResponseEntity<Optional<Transaction>> getTranscation(@PathVariable long id) {
+    Optional<Transaction> transaction = transactionService.getTransaction(id);
+    if (!transaction.isPresent()) {
```

Comment:
You can use Optional.isEmpty method insted of !isPresent

Reply:
Should I add a validationResult object to check the response of the PUT method?